pkg/controller/pv.go

```diff
@@ -101,17 +107,23 @@ func (r *reconcilePV) Reconcile(ctx context.Context, request reconcile.Request)
 		}
 
 		klog.V(5).Infof("Max Retry (%d) for Get AzVolume (%s) exceeded. The CRI is probably deleted.", maxRetry, azVolumeName)
-		r.retryMutex.Lock()
-		delete(r.retryMap, azVolumeName)
-		r.retryMutex.Unlock()
+		r.retryMap.delete(azVolumeName)
 		return reconcile.Result{}, nil
 	}
 
 	if azVolume.Status.Detail != nil {
-		if pv.Status.Phase == corev1.VolumeReleased && azVolume.Status.Detail.Phase == v1alpha1.VolumeBound {
-			updated := azVolume.DeepCopy()
+		updated := azVolume.DeepCopy()
+		if pv.Status.Phase == corev1.VolumeBound {
+			pvClaimName := getNamespacedName(pv.Spec.ClaimRef.Namespace, pv.Spec.ClaimRef.Name)
+			updated.Spec.ClaimName = pvClaimName
```

Comment:
Shouldn't `ClaimName` be part of `Status.Detail` instead of `Spec` since it's updated by the controller in response to a state change of the corresponding `PersistentVolume`. Also, is this field used by anything? I couldn't find any point of usage.

Reply:
It was used prior to adding `claimToVolumeMap` and `volumeToClaimMap`. I must have forgotten to remove it after adding these maps. Thanks for catching this :)